src/git-repo-research-mcp-server/awslabs/git_repo_research_mcp_server/indexer.py

```diff
@@ -0,0 +1,740 @@
+                            repositories.append(metadata_dict['repository_name'])
+                    except Exception as e:
+                        logger.warning(f'Error reading metadata file {metadata_path}: {e}')
+        return repositories
+
+    def load_index(self, repository_name: str) -> Tuple[Optional[Any], Optional[Dict]]:
+        """Load the FAISS index and chunk map for a repository.
+
+        Args:
+            repository_name: Name of the repository
+
+        Returns:
+            Tuple containing:
+            - LangChain FAISS vectorstore if the repository is indexed, None otherwise
+            - Chunk map dictionary if the repository is indexed, None otherwise
+        """
+        index_path = self._get_index_path(repository_name)
+        chunk_map_path = self._get_chunk_map_path(repository_name)
+
+        logger.info(f'Loading index from {index_path}')
+        logger.info(f'Loading chunk map from {chunk_map_path}')
+
+        # Check if the index directory exists
+        if os.path.isdir(index_path):
+            logger.info(f'Index directory exists: {index_path}')
+            # Check if the index files exist
+            index_faiss_path = os.path.join(index_path, 'index.faiss')
+            if not os.path.exists(index_faiss_path):
+                logger.error(f'FAISS index file not found in {index_path}')
+                return None, None
+        else:
+            logger.error(f'Index directory not found: {index_path}')
+            return None, None
+
+        if not os.path.exists(chunk_map_path):
+            logger.error(f'Chunk map not found: {chunk_map_path}')
+            return None, None
+
+        try:
+            # Load the LangChain FAISS index without pickle
+            embedding_function = self.embedding_generator.bedrock_embeddings
+            logger.info(f'Loading FAISS index with embedding function: {embedding_function}')
+
+            try:
+                vector_store = load_index_without_pickle(index_path, embedding_function)
+                logger.info(
+                    f'Successfully loaded vector store with {len(vector_store.docstore._dict)} documents'  # pyright: ignore[reportAttributeAccessIssue]
+                )
+            except Exception as e:
+                logger.error(f'Error loading FAISS index: {e}')
+                # Try a different approach - list the files in the directory
+                logger.info('Files in index directory:')
+                for root, dirs, files in os.walk(index_path):
+                    for file in files:
+                        file_path = os.path.join(root, file)
+                        logger.info(f'  {file_path} ({os.path.getsize(file_path)} bytes)')
+                raise
+
+            # Load the chunk map without pickle
+            chunk_map = load_chunk_map_without_pickle(index_path)
+            logger.info(f'Successfully loaded chunk map with {len(chunk_map["chunks"])} chunks')
```

Comment:
pyright is failing the build here:

```suggestion
            logger.info(f'Successfully loaded chunk map with {len(chunk_map["chunks"]) if chunk_map is not None else 0} chunks')
```

```python
>>> len(None) if None is not None else 0
0
```

Reply:
thank you, added check here